raiden-ts/src/transfers/reducer.ts

```diff
@@ -32,27 +32,25 @@ function transferSecretReducer(
   action: transferSecret | transferSecretRegister.success,
 ): RaidenState {
   const secrethash = action.meta.secrethash;
-  // store when seeing unconfirmed, but registerBlock only after confirmation
+
+  const transferState = state[action.meta.direction][secrethash];
+  if (!transferState) return state;
+
   const registerBlock =
     transferSecretRegister.success.is(action) && action.payload.confirmed
       ? action.payload.txBlock
-      : state[action.meta.direction][secrethash]?.secret?.[1]?.registerBlock ?? 0;
-  // don't overwrite registerBlock if secret already stored with it
-  if (
-    secrethash in state[action.meta.direction] &&
-    state[action.meta.direction][secrethash].secret?.[1]?.registerBlock !== registerBlock
-  )
-    state = {
-      ...state,
-      [action.meta.direction]: {
-        ...state[action.meta.direction],
-        [secrethash]: {
-          ...state[action.meta.direction][secrethash],
-          secret: timed({ value: action.payload.secret, registerBlock }),
-        },
+      : 0;
+
+  return {
+    ...state,
+    [action.meta.direction]: {
+      ...state[action.meta.direction],
+      [secrethash]: {
+        ...transferState,
+        secret: timed({ value: action.payload.secret, registerBlock }),
```

Comment:
This will reset the `ts` timestamp every time an action for this is handled. The `timed` version should only be constructed when the value changes, as it represents when it got to that state. Currently, this timestamp is only used for informative purposes, but we must keep it correct.